contracts/PurchaseHook.sol

```diff
@@ -62,11 +63,18 @@ contract PurchaseHook {
         uint256 /*minKeyPrice*/,
         uint256 /*pricePaid*/
     ) external {
-        console.log("onKeyPurchase");
+        if (msg.sender != address(lock)) {
+            revert Unauthorized();
+        }
+        // console.log("onKeyPurchase");
+        // console.log("recipient %s", recipient);
+        // console.log("referrer %s", referrer);
+        // console.log(referalAmounts[referrer]);
+
         // Do nothing
-        setReferal(recipient, referrer);
-        console.log("referrer", referrer);
-        IPublicLockV13(msg.sender).withdraw(msg.sender, referrer, 1000);
+        setReferrer(recipient, referrer);
+        address currency = lock.tokenAddress();
+        lock.withdraw(currency, referrer, referalAmounts[referrer]); // Pay fee to referrer
```

Comment:
The wirthdraw function takes the currency contract address (if any), the address  that will receive the funds and the amount of funds.